docs/src/main/paradox/scala/http/routing-dsl/testkit.md

```diff
@@ -77,17 +80,88 @@ The following inspectors are defined:
+|`assertMediaType(MediaType expectedType)`                           | Asserts that the media type part of the response's content type matches the given `MediaType`                                       |
+|`assertEntity(String expectedStringContent)`                        | Asserts that the entity data interpreted as UTF8 equals the expected String                                                         |
+|`assertEntityBytes(ByteString expectedBytes)`                       | Asserts that the entity data bytes equal the expected ones                                                                          |
+|`assertEntityAs(Unmarshaller<T> unmarshaller, expectedValue: T)`    | Asserts that the entity data if unmarshalled with the given marshaller equals the given value                                       |
+|`assertHeaderExists(HttpHeader expectedHeader)`                     | Asserts that the response contains an HttpHeader instance equal to the expected one                                                 |
+|`assertHeaderKindExists(String expectedHeaderName)`                 | Asserts that the response contains a header with the expected name                                                                  |
+|`assertHeader(String name, String expectedValue)`                   | Asserts that the response contains a header with the given name and value.                                                          |
+
+It's, of course, possible to use any other means of writing assertions by inspecting the properties the response
+manually. As written above, `TestResponse.entity` and `TestResponse.response` return strict versions of the
+entity data.
+
+## Supporting Custom Test Frameworks
+
+Adding support for a custom test framework is achieved by creating new superclass analogous to
+`JUnitRouteTest` for writing tests with the custom test framework deriving from `akka.http.javadsl.testkit.RouteTest`
+and implementing its abstract methods. This will allow users of the test framework to use `testRoute` and
+to write assertions using the assertion methods defined on `TestResponse`.
+
+
+@@@
+
 ## Testing sealed Routes
 
 The section above describes how to test a "regular" branch of your route structure, which reacts to incoming requests
 with HTTP response parts or rejections. Sometimes, however, you will want to verify that your service also translates
 @ref[Rejections](rejections.md) to HTTP responses in the way you expect.
 
-You do this by wrapping your route with the `akka.http.scaladsl.server.Route.seal`. The `seal` wrapper applies the logic of the in-scope @ref[ExceptionHandler](exception-handling.md) and
+You do this by wrapping your route with the @scala[`akka.http.scaladsl.server.Route.seal`]@java[`akka.http.javadsl.server.Route.seal`]. The `seal` wrapper applies the logic of the in-scope @ref[ExceptionHandler](exception-handling.md) and
```

Comment:
Java does not have the notion of a "in-scope ExceptionHandler", so we need to carry over the Java version here:

I suggest something like this:
```
You do this by calling the `Route.seal()` method. The `Route.seal()` method applies the logic of the @scala[in-scope]
@ref[RejectionHandler](rejections.md#the-rejectionhandler) and @ref[RejectionHandler](rejections.md#the-rejectionhandler)
@java[passed as method arguments] to all exceptions and rejections coming back from the 
route, and translates them to the respective `HttpResponse`.
```